### PR TITLE
blanket try-catch to ensure following unpatch code

### PR DESCRIPTION
**Pull request by williexu**
---
Following from feedback on https://github.com/Azure/azure-cli/pull/6381

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6385
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/azclishell/az_completer.py [hunk 1]
```diff
@@ -205,8 +205,11 @@ def mute_parse_args(self, text):
         AzCliCommandParser.error = error_pass
         AzCliCommandParser._check_value = _check_value_muted
 
-        parse_args = self.argsfinder.get_parsed_args(
-            parse_quotes(text, quotes=False, string=False))
+        try:
+            parse_args = self.argsfinder.get_parsed_args(
```
Comment (troydai) [inline src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/azclishell/az_completer.py]: Same line.

File: src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/azclishell/az_completer.py [hunk 2]
```diff
@@ -205,8 +205,11 @@ def mute_parse_args(self, text):
...
+                parse_quotes(text, quotes=False, string=False))
+        except Exception:  # pylint: disable=broad-except
```
Comment (troydai) [inline src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/azclishell/az_completer.py]: What kind the exception will actually be thrown here?

Comment (williexu) [reply src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/azclishell/az_completer.py]: From what I can see, none. argparse will call `error()` (which is also patched) with any parser related errors. 
This is more of a safety assurance change- we don't want the patched methods to persist anywhere beyond this scope.

Comment (troydai) [reply src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/azclishell/az_completer.py]: Add comments in code.

**Comment by williexu** [comment]
@troydai addressed your comments

**Comment by troydai** [comment]
Merge once CI passes.

**Pull request merged**